cleanbooking.py

```diff
@@ -0,0 +1,97 @@
+import sqlalchemy
+import pandas as pd
+import numpy as np
+import yaml
+from flighttools import flight
+
+file = "***"
+with open(file) as stream:
+    mapper = yaml.safe_load(stream)
+    creds = mapper['hikariConfigs']['middlesex']
+
+db_name = creds['dbname']
+usr_name = creds['username']
+db_password = creds['password']
+
+middlesex_engine = sqlalchemy.create_engine(f'postgresql://{usr_name}:{db_password}@atlas.openlattice.com:30001/{db_name}',connect_args={'sslmode':'require'})
+
+booking_query = "select * from booking;"
+booking_df=pd.read_sql_query(booking_query, middlesex_engine)
+
+# Make a flight object from current yaml
+fl = flight.Flight()
+fl.deserialize('/Users/nicholas/Clients/Middlesex/msobooking.yaml')
+middlesex_booking_fd = fl.schema
+flight_cols = list(fl.get_all_columns())
+
+# Create a dataframe which is a subset of the original table from columns included in the flight
+clean_booking = booking_df[flight_cols]
+
+# Strip all whitespace from object (string) columns, and remove all blank strings ('')
+clean_booking = clean_booking.apply(lambda x: x.str.strip() if x.dtype == "object" else x)
+clean_booking.replace('', np.nan, inplace=True)
+
+# Make float columns to int to fit with the desired datatypes on prod
+clean_booking['SERVED'] = clean_booking['SERVED'].astype('Int64')
+clean_booking['AGE'] = clean_booking['AGE'].astype('Int64')
+clean_booking['SENTENCED_AS_ADULT'] = clean_booking['SENTENCED_AS_ADULT'].map({'Y': 1, 'N': 0}).astype('Int64')
+
+# Map SEX values to new values
+clean_booking['SEX'] = clean_booking['SEX'].map({'M': "Male", "m": "Male", 'F': "Female", "C": "Unknown", "B": "Unknown", "H": "Unknown", "N": "Unknown", "]":"Unknown"})
+
+# Set ambiguous values in RACE to Unknown and set W-White B-Black
+clean_booking.loc[clean_booking['RACE'].isin(['H','U','D1','D3','A','D2','Z','D4','D7','AS','CV','AM']), 'RACE'] = 'Unknown'
```

Comment:
I'm pretty sure the 'A' in race is Asian.

Reply:
modified this